src/askui/utils/annotation_writer.py

```diff
@@ -0,0 +1,271 @@
+
+                const tooltipRect = tooltip.getBoundingClientRect();
+
+                if (tooltipRect.left < imgRect.left) {
+                    tooltip.style.left = "0px";
+                    tooltip.style.transform = "none";
+                }
+
+                if (tooltipRect.right > imgRect.right) {
+                    tooltip.style.left = boxRect.width + "px";
+                    tooltip.style.transform = "translateX(-100%)";
+                }
+            }
+        </script>
+        """
+
+    def _get_elements_html(self) -> str:
+        """
+        Generate HTML for all detected elements.
+
+        Returns:
+            str: Concatenated HTML string for all bounding box elements.
+        """
+        return "".join(self._get_box_html(element) for element in self._elements)
+
+    def _get_box_html(self, element: DetectedElement) -> str:
+        """
+        Generate HTML for a single detected element's bounding box.
+
+        Args:
+            element (DetectedElement): The detected element to generate HTML for.
+
+        Returns:
+            str: HTML string for a single bounding box with tooltip and click
+                handler.
+        """
+        bbox = element.bounding_box
+
+        escaped_text = escape(element.text or "")
+        escaped_name = escape(element.name or "")
+
+        # safe HTML text for tooltips
+        tooltip_text = escaped_name
+        if escaped_text:
+            tooltip_text += f": {escaped_text}"
+
+        style = (
+            f"left:{bbox.xmin}px; "
+            f"top:{bbox.ymin}px; "
+            f"width:{bbox.width}px; "
+            f"height:{bbox.height}px;"
+        )
+
+        return f"""
+        <div
+            class="bbox"
+            style="{style}"
+            data-rawtext="{escaped_text}"
+            onclick="copyTextFromBox(this)"
+            onmouseover="positionTooltip(this)"
+        >
+            <span class="tooltip">{tooltip_text}</span>
+        </div>
+        """
+
+    def _get_full_html(self) -> str:
+        """
+        Generate the complete HTML document with all annotations.
+
+        Returns:
+            str: Complete HTML document as a string including DOCTYPE, head,
+                styles, scripts, and body with annotated image.
+        """
+        return f"""
+            <!DOCTYPE html>
+            <html>
+            <head>
+            <meta charset="UTF-8">
+            <title>Image Annotated By AskUI</title>
+
+            {self._get_style()}
+
+            {self._get_script()}
+
+            </head>
+            <body>
+
+            <div id="copy-message"> 📋 Copied to clipboard! </div>
+
+            <div id="container">
+                <img id="base-image" src="{self._encoded_image}">
+                {self._get_elements_html()}
+            </div>
+
+            </body>
+            </html>
+            """
+
+    def write_to_file(self, output_directory: Path | str) -> Path:
```

Comment:
can we not name it save? Similar like PilImage.save

```suggestion
    def save(self, output_directory: Path | str) -> Path:
```